test/mpi/f77/pt2pt/irsendf.f

```diff
@@ -78,9 +78,10 @@ subroutine test_pair_irsend( comm, errs )
          call MPI_Irsend(send_buf, count, MPI_REAL, next, tag,
      .                   comm, requests(2), ierr)
 C
-         index = -1
-         do while (index .ne. 1)
```

Comment:
Shall we just convert this to a waitall? There doesn't seem to be any logic to using waitany in this case.

Reply:
That or we could wait on the request we care about and call `MPI_Request_free` on the other...